apps/src/templates/SchoolAutocompleteDropdown.jsx

```diff
@@ -0,0 +1,50 @@
+import React, { Component } from 'react';
+import VirtualizedSelect from 'react-virtualized-select';
+import 'react-virtualized/styles.css';
+import 'react-select/dist/react-select.css';
+import 'react-virtualized-select/styles.css';
+
+export default class SchoolAutocompleteDropdown extends Component {
+
+  state = {};
+
+  render() {
+    const getOptions = (q) => {
+      if (!q.length) {
+        return Promise.resolve({ options: [{ value: -1, label: "School not found"}] });
+      }
+      return fetch('/dashboardapi/v1/schoolsearch/' + encodeURIComponent(q) + '/40')
+        .then((response) => response.json())
+        .then((json) => {
+          var schools = [];
+          for (var i = 0; i < json.length; i++) {
+            schools.push({
+              value: json[i].nces_id,
+              label: json[i].name + ' - ' + json[i].city + ', ' + json[i].state + ' ' + json[i].zip
+            });
+          }
+          if (schools.length > 1) {
+            return { options: schools };
+          } else {
+            return { options: [{ value: -1, label: "School not found"}] };
+          }
```

Comment:
It's bascially equivalent, but here's how I would express this block:
```js
// Handle your "no results" special-case with an early-out - especially since it doesn't
// require any of the mapping logic.
if (!json.length) {
  return { options: [{ value: -1, label: "School not found" }] };
}

// Your general case is a good place to use a map function.
return {
  options: json.map(school => ({
    value: school.nces_id,
    label: `${school.name} - ${school.city}, ${school.state} ${school.zip}`
  }))
};
```